### PR TITLE
149 revise documentation

### LINKED ISSUE #149: Revise Documentation
> I came across additional subjects that I believe are worth discussing in order to establish a consistent approach for documenting our functions. However, since these topics are not within the current scope of this issue, I will create a separate issue to address them and potentially establish guidelines.
> 
> #### Functions Documentation
> * `clean_brushedPoints`
>   * No examples
>   * The description is a badge and the title is _Cleans and organizes output to account for NAs and remove empty rows_. Feels like the title should be in description and title should be something concise.
> * `get_dt_rows`
>   * The example is just conceptual on how to use the function, not an actual example that you could run. Although, it is wrapped in `\dontrun{}`.
>   * The description is a badge and the title is _Maps the `lengthMenu`selected value property of DT::datatable to a Shiny variable._ Feels like the title should be in description and the title could be something concise.
>   * Need to add space between the word `lengthMenu` and `selected`.
> * `updateOptionalSelectInput`
>   * Function is exported with no example. Need example.
> * `optionalSliderInput`
>   * Suggestion: If we include lifecycle badge, then add new line after the badge for readability.
>   
> * `panel_group`
>   * The title is _Panel group widget_ with no description other than lifecycle badge. Need more information.
>   * Function is exported with no example. Need example.
> * `panel_item`
>   * The title is _Panel widget_ with no description other than lifecycle badge. Need more information.
>   * Function is exported with no example. Need example.
> * `plot_with_settings_ui`
>   * The title is _UI part of plot-with-settings module_ with no description other than lifecycle badge. Need more information.
>   * Function is exported with no example. Need example.
> * `plot_with_settings_srv`
>   * The title is _Server part of plot-with-settings module_ with no description other than lifecycle badge. Need more information.
>   * Function is exported with no example. Need example.
> * `standard_layout`
>   * Need example.
>   * Enhance title and description
> * `table_with_settings_ui`
>   * Function is exported with no description. Need description.
> * `white_small_well`
>   * The description is a badge. Function is exported with no example. Need example.
> 
> ### Question
> * Do we need lifecycle badge for stable functions? Should we only focus on badges for `experimental` or `deprecated` functions?

_Originally posted by @donyunardi in https://github.com/insightsengineering/teal.widgets/issues/148#issuecomment-1489421749_

### Acceptance Criteria:
* Revise documentation once we agreed on the guideline on how to update the documentation.

### PR DESCRIPTION
**Pull request by m7pr**
Close #149

### CODE REVIEW

**Comment by m7pr** [comment]
Hey @donyunardi I prepared few improvements for https://github.com/insightsengineering/teal.widgets/issues/149. Some comments were outdated, so please check comments in https://github.com/insightsengineering/teal.widgets/issues/149 to see what was omitted. While going through this issue I noticed we could do

- https://github.com/insightsengineering/teal.widgets/issues/227
- https://github.com/insightsengineering/teal.widgets/issues/228
- https://github.com/insightsengineering/teal.widgets/issues/229

At the end of the day I think this was a weird issue, since we tried to provide examples for a lot of utility functions that are used outside of the package to build tmg and tmc modules and encodings panel. So it was hard to create condensed examples, that would focus only on a specific function, which comes with a lot of other functions. Separately some functions just are out of context and do not have much sense. 

Maybe we should create an extra vignette covering those?

File: R/plot_with_settings.R [hunk 1]
```diff
@@ -61,7 +61,7 @@ plot_with_settings_ui <- function(id) {
 #' Plot-with-settings module
 #'
```
Comment (donyunardi) [inline R/plot_with_settings.R:62]: add `@name plot_with_settings`, the same way with `table_with_settings`?

File: R/panel_group.R [hunk 2]
```diff
@@ -35,9 +73,9 @@ panel_group <- function(..., id = NULL) {
 }
 
 #' Panel widget
```
Comment (donyunardi) [inline R/panel_group.R:75]:
```suggestion
#' Panel item widget
```

**Review by donyunardi — COMMENTED** [review]
Thanks @m7pr!
Just small comments and this is good to go!

File: R/plot_with_settings.R (hunk 1 above)
Comment (m7pr) [reply R/plot_with_settings.R:63]: 2d3c73b3db317fcf4886c21fac3c0e350b8c7aec

**Comment by m7pr** [comment]
Changes incorporated @donyunardi . Take a look, when you have some spare time.

**Review by donyunardi — APPROVED** [review]
LGTM!
And thanks for creating new issue about the panel widget so we can move forward with this PR.

**Pull request merged**